node/cmd/guardiand/node.go

```diff
@@ -626,7 +629,14 @@ func runNode(cmd *cobra.Command, args []string) {
 		logger.Fatal("Please specify --nodeKey")
 	}
 	if *guardianKeyPath == "" {
-		logger.Fatal("Please specify --guardianKey")
+		if *guardianSignerUri == "" {
+			logger.Fatal("Please specify --guardianKey or --guardianSignerUri")
+		}
+	} else {
+		// If guardianKeyPath is set, set guardianSignerUri to the file signer URI, pointing to guardianKeyPath.
+		// This ensures that the signer-abstracted guardian has backwards compatibility with guardians that would
+		// just like to ignore the new guardianSignerUri altogether.
+		*guardianSignerUri = fmt.Sprintf("file://%s", *guardianKeyPath)
```

Comment:
My preference would be to use https://pkg.go.dev/net/url#URL.JoinPath instead of Sprintf so that the program can help with repeated characters, relative filepaths and so on.

Reply:
@johnsaigle  I'm going to leave the `Sprintf` as is, since I feel like an incorrect path should just lead to failure and we aren't really doing that now with the `guardianKeyPath` anyway.

@banescusebi My intention is that the `guardianKeyPath` gets preference, so leaving this as is for now.

Reply:
Yes I'm reverting back to nested if.